spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -85,67 +242,167 @@ def write_recording(recording, save_path, acquisition_name='ElectricalSeries', *
-            nwbfile_kwargs['identifier'] = str(uuid.uuid4())
-        input_nwbfile_kwargs = {
-            'session_start_time': datetime.now()}
-        input_nwbfile_kwargs.update(nwbfile_kwargs)
-        nwbfile = NWBFile(**input_nwbfile_kwargs)
-
-        device = nwbfile.create_device(name='device_name')
-        eg_name = 'electrode_group_name'
-        eg_description = "electrode_group_description"
-        eg_location = "electrode_group_location"
-
-        electrode_group = nwbfile.create_electrode_group(
-            name=eg_name,
-            location=eg_location,
-            device=device,
-            description=eg_description
-        )
+            read_mode = 'r+'
+        else:
+            read_mode = 'w'
 
-        for m in range(M):
-            location = recording.get_channel_property(m, 'location')
-            impedence = -1.0
-            while len(location) < 3:
-                location = np.append(location, [0])
-            nwbfile.add_electrode(
-                id=m,
-                x=float(location[0]), y=float(location[1]), z=float(location[2]),
-                imp=impedence,
-                location='electrode_location',
-                filtering='none',
-                group=electrode_group,
-            )
-        electrode_table_region = nwbfile.create_electrode_table_region(
-            list(range(M)),
-            'electrode_table_region'
-        )
+        with NWBHDF5IO(save_path, mode=read_mode) as io:
+            if read_mode == 'r+':
+                nwbfile = io.read()
+            else:
+                kwargs = {'session_description': 'No description',
+                          'identifier': str(uuid.uuid4()),
+                          'session_start_time': datetime.now()}
+                kwargs.update(**nwbfile_kwargs)
+                nwbfile = NWBFile(**kwargs)
 
-        rate = recording.get_sampling_frequency()
-        ephys_data = recording.get_traces().T
-
-        ephys_ts = ElectricalSeries(
-            name=acquisition_name,
-            data=ephys_data,
-            electrodes=electrode_table_region,
-            starting_time=recording.frame_to_time(0),
-            rate=rate,
-            resolution=1e-6,
-            comments='Generated from SpikeInterface::NwbRecordingExtractor',
-            description='acquisition_description'
-        )
+            # Tests if Device already exists
+            aux = [isinstance(i, Device) for i in nwbfile.children]
+            if any(aux):
+                device = nwbfile.children[np.where(aux)[0][0]]
+            else:
+                device = nwbfile.create_device(name='Device')
+
+            # Check if 'groups' property exist for channels in Recording
+            if 'group' in recording.get_shared_channel_property_names():
+                el_groups_names = np.unique(recording.get_channel_groups()).tolist()
+                # Creates electrode groups for groups names in Recording
```

Comment:
corrected:
4a2680819e9a1881bcc1770596dd7b2621a567ef

Reply:
That's not what I meant. What I mean is these lines:
```
               for grp_name in el_groups_names:
                    if str(grp_name) not in nwbfile.electrode_groups:
                        elec_group = nwbfile.create_electrode_group(
                            name=str(grp_name),
                            location="electrode_group_location",
                            device=device,
                            description="electrode_group_description"
                        )
            else:
                if "0" not in nwbfile.electrode_groups:
                    elec_group = nwbfile.create_electrode_group(
                        name="0",
                        location="electrode_group_location",
                        device=device,
                        description="electrode_group_description"
                    )
```
contain redundant code because it calls `nwbfile.create_electrode_group` with nearly the same args. Instead we could have something like:
```
if 'group' in recording.get_shared_channel_property_names():
    el_groups_names = np.unique(recording.get_channel_groups()).tolist()
else:
    el_group_names = ["0"]
```
and then we just need to have a single `nwbfile.create_electrode_group` call in the subsequent code
```